src/lib/planviz/lqp_visualizer.cpp

```diff
@@ -46,47 +48,70 @@ void LQPVisualizer::_build_subtree(const std::shared_ptr<AbstractLQPNode>& node,
     _build_subtree(right_input, visualized_nodes);
     _build_dataflow(right_input, node);
   }
+
+  if (const auto projection = std::dynamic_pointer_cast<ProjectionNode>(node)) {
+    for (const auto& column_expression : projection->column_expressions()) {
+      if (column_expression->is_subselect()) {
+        _build_subtree(column_expression->subselect_node(), visualized_nodes);
+
+        auto edge_info = _default_edge;
+        edge_info.label = "Scalar Subquery";
+        edge_info.style = "dashed";
+        _build_dataflow(column_expression->subselect_node(), node, edge_info);
```

Comment:
Maybe I am missing something but why don't you just call _add_edge here directly?
The current way _build_dataflow is implemented is still ugly, imo

Reply:
Yea, good point. I should have seen this straight away.